test/unit/ErrorEstimationTest.cpp

```diff
@@ -0,0 +1,132 @@
+#include <cppunit/TestFixture.h>
+#include <cppunit/extensions/HelperMacros.h>
+
+#include "coupling/CouplingMDDefinitions.h"
+#include "coupling/ErrorEstimation.h"
+
+/** tests several properties of the ErrorEstimation. 
+ *  @author Vahid Jafarimann
+ */
+class ErrorEstimationTest : public CppUnit::TestFixture {
+  CPPUNIT_TEST_SUITE(ErrorEstimationTest);
+  CPPUNIT_TEST(testVelocityError);
+  CPPUNIT_TEST(testDensityError);
+  CPPUNIT_TEST(testTemperatureError);
+  CPPUNIT_TEST(testPressureError);
+  CPPUNIT_TEST_SUITE_END();
+
+public:
+  void testVelocityError() {
+    double velocity = 5;
+    double temperature = 18;
+    double numberOfParticle = 16;
+    double particleMass = 2.0;
+    double soundSpeed = 1.0;
+    double numberOfSamples = 9;
+    double cellVolume = 1.0;
+    std::cout << "Run velocityError test..." << std::endl;
+    _testVelocityError(velocity, temperature, numberOfParticle, particleMass, soundSpeed, numberOfSamples, cellVolume);
+  }
+  
+  void testDensityError() {
+    double velocity = 5;
+    double temperature = 16;
+    double numberOfParticle = 25;
+    double particleMass = 1.667;
+    double soundSpeed = 1.0;
+    double numberOfSamples = 64;
+    double cellVolume = 1.0;
+    std::cout << "Run densityError test..." << std::endl;
+    _testDensityError(velocity, temperature, numberOfParticle, particleMass, soundSpeed, numberOfSamples, cellVolume);
+  }
+  
+  void testTemperatureError() {
+    double velocity = 5;
+    double temperature = 16;
+    double numberOfParticle = 66.78688306;
+    double particleMass = 1.667;
+    double soundSpeed = 1.0;
+    double numberOfSamples = 100;
+    double cellVolume = 1.0;
+    std::cout << "Run temperatureError test..." << std::endl;
+    _testTemperatureError(velocity, temperature, numberOfParticle, particleMass, soundSpeed, numberOfSamples, cellVolume);
+  }
+  
+  void testPressureError() {
+    double velocity = 5;
+    double temperature = 16;
+    double numberOfParticle = 25;
+    double particleMass = 1.667;
+    double soundSpeed = 1.0;
+    double numberOfSamples = 1.667;
+    double cellVolume = 1.0;
+    std::cout << "Run pressureError test..." << std::endl;
+    _testPressureError(velocity, temperature, numberOfParticle, particleMass, soundSpeed, numberOfSamples, cellVolume);
+  }
+
+
+private:
+  void _testVelocityError(double velocity, double temperature, double numberOfParticle, double particleMass, double soundSpeed, double numberOfSamples, double cellVolume) {
+      
+    coupling::error::ErrorEstimation errorControl(velocity, temperature, numberOfParticle, particleMass, soundSpeed, numberOfSamples, cellVolume);
+    double errorVelo = errorControl.getError(coupling::error::ErrorEstimation::Velocity, coupling::error::ErrorEstimation::Absolute);
+    CPPUNIT_ASSERT(errorVelo == 0.25);
```

Comment:
Why are the expected values hardcoded here instead of also being passed into the function as a parameter? They seem to depend on the concrete values, right? (also see `_testDensityError(...)`)